server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DirectResolutionRemoteDeliveryIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+import org.apache.james.utils.SMTPMessageSender;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.collect.ImmutableList;
+
+public class DirectResolutionRemoteDeliveryIntegrationTest {
+    private static final String JAMES_ANOTHER_DOMAIN = "james.com";
+    private static final String JAMES_ANOTHER_MX_DOMAIN_1 = "mx1.james.com";
+    private static final String JAMES_ANOTHER_MX_DOMAIN_2 = "mx2.james.com";
+    private static final List<String> JAMES_ANOTHER_MX_DOMAINS = ImmutableList.of(JAMES_ANOTHER_MX_DOMAIN_1, JAMES_ANOTHER_MX_DOMAIN_2);
+
+    private static final String FROM = "from@" + DEFAULT_DOMAIN;
+    private static final String RECIPIENT = "touser@" + JAMES_ANOTHER_DOMAIN;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Rule
+    public IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+    @Rule
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+    @Rule
+    public FakeSmtp fakeSmtp = new FakeSmtp();
+
+    private TemporaryJamesServer jamesServer;
+    private DataProbe dataProbe;
+
+    @Before
+    public void setup() {
+        fakeSmtp.awaitStarted(awaitAtMostOneMinute);
+    }
+
+    @After
+    public void tearDown() {
+        if (jamesServer != null) {
+            jamesServer.shutdown();
+        }
+    }
+
+    @Test
+    public void directResolutionShouldBeWellPerformed() throws Exception {
+        InMemoryDNSService inMemoryDNSService = new InMemoryDNSService()
+            .registerMxRecord(JAMES_ANOTHER_DOMAIN, fakeSmtp.getContainer().getContainerIp());
+
+        jamesServer = TemporaryJamesServer.builder()
+            .withBase(SMTP_ONLY_MODULE)
+            .withOverrides(binder -> binder.bind(DNSService.class).toInstance(inMemoryDNSService))
+            .withMailetContainer(MailetContainer.builder()
+                .putProcessor(CommonProcessors.simpleRoot())
+                .putProcessor(CommonProcessors.error())
+                .putProcessor(directResolutionTransport())
+                .putProcessor(CommonProcessors.bounces()))
+            .build(temporaryFolder);
+
+        dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+        dataProbe.addUser(FROM, PASSWORD);
+
+        messageSender.connect(LOCALHOST_IP, SMTP_PORT)
+            .sendMessage(FROM, RECIPIENT);
+
+        awaitAtMostOneMinute.until(this::messageIsReceivedByTheSmtpServer);
+    }
+
+    @Test
+    public void directResolutionShouldFailoverOnSecondMxWhenFirstMxFailed() throws Exception {
+        InMemoryDNSService inMemoryDNSService = new InMemoryDNSService()
+            .registerRecord(JAMES_ANOTHER_DOMAIN, ImmutableList.of(), JAMES_ANOTHER_MX_DOMAINS, ImmutableList.of())
+            .registerMxRecord(JAMES_ANOTHER_MX_DOMAIN_1, LOCALHOST_IP) // TODO find another IP address
```

Comment:
You can introduce a second FakeSmtp rule.

Reply:
Or launch a container that do not listen on port 25 ?

Reply:
you should choose an unroutable address like 10.x or 192.168.x